spec/index.md

```diff
@@ -1299,77 +1303,82 @@ In the reply callback of a [inter-canister method call](#system-api-call), the a
+
+    ic0.msg_method_name_size : () -> I;                                                   // F
+    ic0.msg_method_name_copy : (dst : I, offset : I, size : I) -> ();                     // F
+    ic0.accept_message : () -> ();                                                        // F
+
+    ic0.call_new :
+      ( callee_src  : I,
+        callee_size : I,
+        name_src  : I,
+        name_size : I,
         reply_fun : i32,
         reply_env : i32,
         reject_fun : i32,
         reject_env : i32
-      ) -> ();
-    ic0.call_on_cleanup : (fun : i32, env : i32) -> ();                         // U CQ Ry Rt CRy CRt T
-    ic0.call_data_append : (src : i32, size : i32) -> ();                       // U CQ Ry Rt CRy CRt T
-    ic0.call_cycles_add : (amount : i64) -> ();                                 // U Ry Rt T
-    ic0.call_cycles_add128 : (amount_high : i64, amount_low: i64) -> ();        // U Ry Rt T
-    ic0.call_perform : () -> ( err_code : i32 );                                // U CQ Ry Rt CRy CRt T
-
-    ic0.stable_size : () -> (page_count : i32);                                 // * s
-    ic0.stable_grow : (new_pages : i32) -> (old_page_count : i32);              // * s
-    ic0.stable_write : (offset : i32, src : i32, size : i32) -> ();             // * s
-    ic0.stable_read : (dst : i32, offset : i32, size : i32) -> ();              // * s
-    ic0.stable64_size : () -> (page_count : i64);                               // * s
-    ic0.stable64_grow : (new_pages : i64) -> (old_page_count : i64);            // * s
-    ic0.stable64_write : (offset : i64, src : i64, size : i64) -> ();           // * s
-    ic0.stable64_read : (dst : i64, offset : i64, size : i64) -> ();            // * s
-
-    ic0.certified_data_set : (src: i32, size: i32) -> ();                       // I G U Ry Rt T
-    ic0.data_certificate_present : () -> i32;                                   // *
-    ic0.data_certificate_size : () -> i32;                                      // Q CQ
-    ic0.data_certificate_copy : (dst: i32, offset: i32, size: i32) -> ();       // Q CQ
-
-    ic0.time : () -> (timestamp : i64);                                         // *
-    ic0.global_timer_set : (timestamp : i64) -> i64;                            // I G U Ry Rt C T
-    ic0.performance_counter : (counter_type : i32) -> (counter : i64);          // * s
-    ic0.is_controller: (src: i32, size: i32) -> ( result: i32);                 // * s
-    ic0.in_replicated_execution: () -> (result: i32);                           // * s
-
-    ic0.debug_print : (src : i32, size : i32) -> ();                            // * s
-    ic0.trap : (src : i32, size : i32) -> ();                                   // * s
+      ) -> ();                                                                            // U CQ Ry Rt CRy CRt T
+    ic0.call_on_cleanup : (fun : i32, env : i32) -> ();                                   // U CQ Ry Rt CRy CRt T
```

Comment:
From interface spec meeting: it may be worth generalizing the `env` to `I` as well since it could legitimately be a pointer to memory.

Reply:
Ohh, good point. cdk seems to be using it by putting raw pointers there...

Reply:
seems like tables are indexed by u32 in both wasm32 and wasm64 modes

Reply:
I'm working on experimenting wasm64 support in Rust CDK.
And I just noticed that it is not only `env` but also the `fun`/`reply_fun`/`reject_fun` will be 32 bit in wasm64.

The `env: i32` can be handled by introduced an [indirection](https://github.com/dfinity/cdk-rs/pull/492).
When it comes to `fun`, I don't know how to do it. In [current Rust CDK](https://github.com/dfinity/cdk-rs/blob/05ca5719c3baf7fb4c4982e31097ff8b07fd8a25/src/ic-cdk/src/api/call.rs#L106-L114), we are passing function pointers.